### DISCUSSION TITLE
Python

### DISCUSSION THREAD

**Discussion by igorek1105** [community/community#23334] [New to GitHub]
<p>Greetings from Russia,<br>
My name is Igor, I would like to learn how to program in python, but I don’t know where to start :frowning: Advise something to learn: websites, books, courses, anything. I have already studied arrays, lists, in other words, the basics. But I do not know how to move on.<br>
Please help, thank you in advance<br>
Igor</p>

**Comment by Wabri** [comment, marked as answer]
<p>Hi <a class="mention" href="/igorek1105">@igorek1105</a>,</p>
<p>Welcome to the Github Community Forum!</p>
<p>There are several courses and resources where you can learn pyhton:</p>
<ul>
<li>
<strong>Video courses</strong> :
<ul>
<li>Udemy, the course created by <a href="https://www.udemy.com/python-the-complete-python-developer-course/#instructor-3" rel="nofollow noopener">Tim Buchalka’s</a> one of the best teacher that i had in my life -&gt; <a href="https://www.udemy.com/python-the-complete-python-developer-course/" rel="nofollow noopener">python-the-complete-python-developer-course</a>
</li>
<li>Coursera -&gt; <a href="https://www.codecademy.com/learn/learn-python-3" rel="nofollow noopener">learn-python-3</a>
</li>
<li>Sololearn -&gt; <a href="https://www.sololearn.com/Course/Python/" rel="nofollow noopener">Python course</a>
</li>
<li>YouTube -&gt; <a href="https://www.youtube.com/watch?v=rfscVS0vtbw" rel="nofollow noopener">Learn Python - Full Course for Beginners</a>
</li>
</ul>
</li>
<li>
<strong>Website</strong> :
<ul>
<li>This huge list of things -&gt; <a href="https://www.quora.com/How-should-I-start-learning-Python-1" rel="nofollow noopener">Start-learning-Python</a>
</li>
<li>Obviously the python docs -&gt; <a href="https://docs.python.org/3/tutorial/index.html" rel="nofollow noopener">docs python 3.7</a>
</li>
<li>This huge python script file example -&gt; <a href="https://learnxinyminutes.com/docs/python3/" rel="nofollow noopener">learn x in y minutes python3</a>
</li>
</ul>
</li>
<li>
<strong>Books</strong> :
<ul>
<li>The best book O’Reilly publisher for pyhton -&gt; <a href="http://shop.oreilly.com/product/0636920028154.do" rel="nofollow noopener">Learning Python, 5th Edition Powerful Object-Oriented Programming</a>
</li>
<li>The best hands on book of python is without any doubts -&gt; <a href="https://automatetheboringstuff.com/" rel="nofollow noopener">Automate the boring staff</a>
</li>
</ul>
</li>
<li>
<strong>Anything</strong> :
<ul>
<li>I’m sometimes start a new repository to automate my computer operations using python and if you need to do some exercise you can start help me: <a href="https://github.com/Wabri/i3battery" rel="nofollow noopener">i3battery</a>, <a href="https://github.com/YetAnotherPackageInstaller/YAPI" rel="nofollow noopener">YAPI-origin</a>, <a href="https://github.com/YetAnotherPackageInstaller/rewrite" rel="nofollow noopener">YAPI-rewrite</a>, and more…</li>
</ul>
</li>
</ul>
<p>Ask if you need more advices :slight_smile:</p>
<p>-Gabriele-</p>
<p>Mark helpfull posts with Accept as Solution to help other users locate important info. Don’t forget to give Kudos for great contents!</p>

**Comment by igorek1105** [comment]
<p>Gabriele, thank you so much!</p>

**Comment by satyam41** [comment]
<p>you should visit youtube channel codewithharry</p>
<p>you will got all about python</p>

**Comment by kushnertodd** [comment]
This is an Introduction to Python Jupyter notebook to teach Python with little or no prior experience in programming.

http://www.toddkushnerllc.com/intro_to_python.html

**Comment by Kill0geR** [comment]
The best way to learn programming is to learn it on this site: https://www.freecodecamp.org/
